Adding RLP decoding utility methods to runtime stdlib (#1397)
onflow/cadence: docs/language/built-in-functions.md

  Follow [best practices](https://github.com/ConsenSys/smart-contract-best-practices/blob/051ec2e42a66f4641d5216063430f177f018826e/docs/recommendations.md#remember-that-on-chain-data-is-public)
  to prevent security issues when using this function.

> ## rlpDecodeString

Reviewer: We want to make namespace the RLP functionality by exposing it as functions in a contract, e.g. `RLP.decodeString` here. We can do that in a follow-up PR though and can leave the current implementation as-is, to avoid making this PR bigger.

At least we should keep the naming consistent with existing global functionality like BLS (e.g. `AggregateBLSSignatures`), i.e. rename this function to `DecodeRLPString` for now.

(See discussion https://axiomzen.slack.com/archives/CG0B7CJAJ/p1644270003698399)

Author: renamed them for now until naming is finalized.

`cadence•fun rlpDecodeString(input: [UInt8]): [UInt8]`

>   Accepts a RLP encoded byte array (called string in the context of RLP) and decodes it. 
>   Input should only contain a single encoded value for an string; if the encoded value type doesn't match or it has trailing unnecessary bytes it would error out. Any error while decoding fails the transaction. 

Reviewer: ```suggestion
  Decodes an RLP-encoded byte array (called string in the context of RLP). 
  The byte array should only contain of a single encoded value for a string; if the encoded value type does not match, or it has trailing unnecessary bytes, the program aborts.
  If any error is encountered while decoding, the program aborts.
```

> ## rlpDecodeList

Reviewer: Same as above, please rename to `DecodeRLPList` for now to keep naming consistent.

Author: done

`cadence•fun rlpDecodeList(input: [UInt8]): [[UInt8]]`

>   Accepts a RLP encoded list and decodes it into a array of encoded items.
>   Input should only contain a single encoded value for a list; if the encoded value type doesn't match or it has trailing unnecessary bytes it would error out. Any error while decoding fails the transaction. 

Reviewer: ```suggestion
  Decodes an RLP-encoded list (called string in the context of RLP) into an array of RLP-encoded items.
  Note that this function does not recursively decode, so each element of the resulting array is RLP-encoded data.
  The byte array should only contain of a single encoded value for a list; if the encoded value type does not match, or it has trailing unnecessary bytes, the program aborts.
  If any error is encountered while decoding, the program aborts.
```
